src/UriNormalize.c

```diff
@@ -245,6 +247,24 @@ static URI_INLINE void URI_FUNC(LowercaseInplace)(const URI_CHAR * first,
 
 
 
+static URI_INLINE void URI_FUNC(LowercaseInplaceAfterPercentEncoding)(const URI_CHAR * first,
+		const URI_CHAR * afterLast) {
+	if ((first != NULL) && (afterLast != NULL) && (afterLast > first)) {
+		URI_CHAR * i = (URI_CHAR *)first;
+		const int lowerUpperDiff = (_UT('a') - _UT('A'));
+		URI_CHAR * skipLowercasingUntilPosition = NULL;
+		for (; i < afterLast; i++) {
+			if ((*i >= _UT('A')) && (*i <=_UT('Z')) && skipLowercasingUntilPosition < i) {
+				*i = (URI_CHAR)(*i + lowerUpperDiff);
+			} else if (*i == _UT('%')) {
+				skipLowercasingUntilPosition = i + 2;
```

Comment:
I'm not deeply familiar with percent-encodings... So I don't know if it's possible that there is a `%` after another `%`. If it's not (and 2 characters after a `%` sign can always be skipped), then I believe only `i` could be incremented by 3.

Reply:
@kocsismate I believe I found part of the answer to that here:

> [2.4](https://datatracker.ietf.org/doc/html/rfc3986#section-2.4).  When to Encode or Decode
> [..]
> Because the percent ("%") character serves as the indicator for percent-encoded octets, it must be percent-encoded as "%25" for that octet to be used as data within a URI.